google-ads-migration-examples/src/main/java/com.google.ads.googleads.migration/createcampaign/CreateCompleteCampaignAdWordsApiOnly.java

```diff
@@ -0,0 +1,429 @@
+import com.google.api.ads.adwords.axis.v201809.cm.Bids;
+import com.google.api.ads.adwords.axis.v201809.cm.Budget;
+import com.google.api.ads.adwords.axis.v201809.cm.BudgetBudgetDeliveryMethod;
+import com.google.api.ads.adwords.axis.v201809.cm.BudgetOperation;
+import com.google.api.ads.adwords.axis.v201809.cm.BudgetReturnValue;
+import com.google.api.ads.adwords.axis.v201809.cm.BudgetServiceInterface;
+import com.google.api.ads.adwords.axis.v201809.cm.Campaign;
+import com.google.api.ads.adwords.axis.v201809.cm.CampaignOperation;
+import com.google.api.ads.adwords.axis.v201809.cm.CampaignReturnValue;
+import com.google.api.ads.adwords.axis.v201809.cm.CampaignServiceInterface;
+import com.google.api.ads.adwords.axis.v201809.cm.CampaignStatus;
+import com.google.api.ads.adwords.axis.v201809.cm.CpcBid;
+import com.google.api.ads.adwords.axis.v201809.cm.ExpandedTextAd;
+import com.google.api.ads.adwords.axis.v201809.cm.Keyword;
+import com.google.api.ads.adwords.axis.v201809.cm.KeywordMatchType;
+import com.google.api.ads.adwords.axis.v201809.cm.Money;
+import com.google.api.ads.adwords.axis.v201809.cm.NetworkSetting;
+import com.google.api.ads.adwords.axis.v201809.cm.Operator;
+import com.google.api.ads.adwords.axis.v201809.cm.UrlList;
+import com.google.api.ads.adwords.axis.v201809.cm.UserStatus;
+import com.google.api.ads.adwords.lib.factory.AdWordsServicesInterface;
+import com.google.api.ads.common.lib.conf.ConfigurationLoadException;
+import com.google.api.ads.common.lib.exception.OAuthException;
+import com.google.api.ads.common.lib.exception.ValidationException;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.ads.adwords.lib.client.AdWordsSession;
+import com.google.api.ads.common.lib.auth.OfflineCredentials;
+import com.google.api.ads.common.lib.auth.OfflineCredentials.Api;
+import java.io.UnsupportedEncodingException;
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.net.URLEncoder;
+
+import org.joda.time.DateTime;
+
+import static com.google.api.ads.common.lib.utils.Builder.DEFAULT_CONFIGURATION_FILENAME;
+
+/**
+ * This code example is the first in a series of code examples that shows how to create a Search
+ * campaign using the AdWords API, and then migrate it to the Google Ads API one functionality at a
+ * time. See other examples for code examples in various stages of migration.
+ *
+ * This code example represents the initial state, where the AdWords API is used to create a
+ * campaign budget, a Search campaign, ad groups, keywords and expanded text ads. None of the
+ * functionality has yet been migrated to the Google Ads API.
+ */
+public class CreateCompleteCampaignAdWordsApiOnly {
+
+  private static final int NUMBER_OF_ADS = 5;
+
+  private static final List<String> KEYWORDS_TO_ADD = Arrays.asList("mars cruise", "space hotel");
+
+  public static void main(String[] args) {
```

Comment:
Good idea, although let's make it optional and only fail if it is missing from the config file.

Reply:
@nwbirnie @jradcliff this is not how the examples in the [AdWords v201809](https://github.com/googleads/googleads-java-lib/tree/master/examples/adwords_axis/src/main/java/adwords/axis/v201809) are structured. Is it better to provide this option or preserve consistency? I was going for consistency given that it may be common to look at the existing examples while using this set of examples.

Reply:
The AdWordsApiOnly example (this file) is a special case in that it does not use the Google Ads API at all. However, for every other case, you'll need the customer ID since, unlike `AdWordsSession`, `GoogleAdsApiClient` is _not_ bound to a customer ID. I see 2 options:
1. Pull the client customer ID from the `AdWordsSession`. Downside here is this won't work for the final Google Ads API-only example, where you won't have an `AdWordsSession`.
2. Require customer ID in every example. Downside here is that it's technically not required for _this_ example if the user happened to specify the customer ID in their config file. However, it _is_ required for every other example, and making it a command-line param also ensures the example will work even if the user has not specified client customer ID in the AdWords API section of their config file.
Given the above, I vote for option 2 and (in spite of my previous comment :smile: ) making customer ID _required_ in all cases. You'll then have to modify your `AdWordsSession` creation to include a call to `setClientCustomerId(params.customerId`.

Reply:
Done.